win32ss/user/winsrv/consrv/frontends/gui/conwnd.c

```diff
@@ -2135,6 +2139,51 @@ OnMove(PGUI_CONSOLE_DATA GuiData)
     GuiData->GuiInfo.WindowOrigin.y = rcWnd.top;
 }
 
+static VOID
+OnDropFiles(HWND hWnd, PCONSRV_CONSOLE Console, HDROP hDrop)
+{
+    WCHAR *pch, szPath[MAX_PATH + 2];
+    INPUT_RECORD ir;
+    USHORT vk;
+
+    szPath[0] = L'"';
+    DragQueryFileW(hDrop, 0, &szPath[1], ARRAYSIZE(szPath) - 1);
+    if (wcschr(&szPath[1], L' ') != NULL)
+    {
+        StringCchCatW(szPath, ARRAYSIZE(szPath), L"\"");
+        pch = szPath;
+    }
+    else
+    {
+        pch = &szPath[1];
+    }
+
+    ir.EventType = KEY_EVENT;
```

Comment:
You can now call `PasteText(Console, pch, wcslen(pch));` (instead of duplicating pasting code).
Also I would suggest changing "pch" to "pszPath".

Reply:
Where is PasteText function? Not applied yet.

Reply:
OK.